### DISCUSSION TITLE
How to bind mouse right-click with Copy & Paste? (Like windows terminal)

### DISCUSSION THREAD

**Discussion by lentikr** [wezterm/wezterm#3541] [Q&A] [answered]
As ↑.

**Comment by yutkat** [comment]
```lua
local wezterm = require 'wezterm'
local act = wezterm.action
local config = {}

config.mouse_bindings = {
	{
		event = { Down = { streak = 1, button = "Right" } },
		mods = "NONE",
		action = act({ PasteFrom = "Clipboard" }),
	},
}

return config
```

https://wezfurlong.org/wezterm/config/mouse.html?h=bind#configuring-mouse-assignments

**Reply by lentikr** [reply]
When I add this to my config, select some text and right-click the mouse, the selected text automaticaly duplicated, but I just want to copy in that case.

**Reply by yutkat** [reply]
Is this what you mean?

```lua
local wezterm = require 'wezterm'
local act = wezterm.action
local config = {}

config.mouse_bindings = {
	{
		event = { Down = { streak = 1, button = "Right" } },
		mods = "NONE",
		action = act({ CopyTo = "Clipboard" }),
	},
}

return config
```

**Reply by lentikr** [reply]
I need both of these. I think it would be better to be able to judge when right-clicking, and perform a copy operation if there is some text currently selected, and if not, perform a paste. But I don't know how to implement this judgement.

**Reply by yutkat** [reply, marked as answer]
If you select (not wezterm copy mode), you want to copy, and if you don't select anything, you want to paste?

```lua
local wezterm = require("wezterm")
local act = wezterm.action
local config = {}

config.mouse_bindings = {
	{
		event = { Down = { streak = 1, button = "Right" } },
		mods = "NONE",
		action = wezterm.action_callback(function(window, pane)
			local has_selection = window:get_selection_text_for_pane(pane) ~= ""
			if has_selection then
				window:perform_action(act.CopyTo("ClipboardAndPrimarySelection"), pane)
				window:perform_action(act.ClearSelection, pane)
			else
				window:perform_action(act({ PasteFrom = "Clipboard" }), pane)
			end
		end),
	},
}

return config
```

**Reply by lentikr** [reply]
That's what I need. Thanks a lot.

**Reply by highpost** [reply]
I was just looking for this!

**Reply by hongwen000** [reply]
Just mention, for beginners like me: Remember to load wezterm.action to the script! 
`local act = wezterm.action`

**Reply by nigerninja** [reply]
Thanks!

**Reply by CosmicBagel** [reply]
This is gold, thank you!

**Reply by itripley** [reply]
where do i add it?

**Reply by shunia** [reply]
> where do i add it?

You should already find it out but here you go: https://wezterm.org/config/files.html#configuration-files
